src/core/qgsdistancearea.cpp

```diff
@@ -537,6 +540,111 @@ double QgsDistanceArea::measureLine( const QgsPoint &p1, const QgsPoint &p2, Qgs
   return result;
 }
 
+double QgsDistanceArea::measureLineProjected( const QgsPoint &p1, double distance, double azimuth, QgsPoint *projected_point )
+{
+  double result = 0.0;
+  QgsPoint p2;
+  if ( geographic() )
```

Comment:
Also just wanted to add that I think this is an important conversation to raise - I've seen many plugins (including processing) incorrectly use QgsDistanceArea due to missing one of the steps required to get a properly initialized object ready for measurements.